packages/elastic-safer-lodash-set/index.d.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is forked from the DefinitlyTyped project (https://github.com/DefinitelyTyped/DefinitelyTyped),
+ * and may include modifications made by Elasticsearch B.V.
+ * Elasticsearch B.V. licenses this file to you under the MIT License.
+ * See `packages/elastic-safer-lodash-set/LICENSE` more information.
+ */
+
+export = SaferLodashSet;
+export as namespace SaferLodashSet;
+
+type Many<T> = T | readonly T[];
+type PropertyName = string | number | symbol;
+type PropertyPath = Many<PropertyName>;
+type SetWithCustomizer<T> = (nsValue: any, key: string, nsObject: T) => any;
+
+declare const SaferLodashSet: SaferLodashSet.SaferLoDashStatic;
+declare namespace SaferLodashSet {
+  interface SaferLoDashStatic {
+    /**
+     * Sets the value at path of object. If a portion of path doesn’t exist it’s
+     * created. Arrays are created for missing index properties while objects
+     * are created for all other missing properties. Use SaferLodashSet.setWith
+     * to customize path creation.
+     *
```

Comment:
Probably a naive question, but did we tried to discuss with upstream whereas they would be interested with this behavioral change (or not) before forking these methods?

Reply:
I'm not sure about others, but I tried reaching out about `set` behavioral changes approximately 8-10 months ago through various means and was given the response that the lodash author essentially considers `set`s current behavior (as well as `zipObjectDeep` which just made a big splash recently) to be intentional. Pretty sure given some of the past pushback on safety considerations from upstream patches to `lodash` that these changes would not be welcomed, but I could be wrong 🤷

Reply:
When I started this PR we were still running on a combination of our own internal fork of Lodash 3 while other parts of our code was using Lodash 4 - so to get this patch into Kibana ASAP, I went with this approach to begin with.

That being said, I will try to get this merged upstream afterwards, possibly with a flag to turn it on in Lodash 4 so that we can get rid of this "hack". That would be preferable :)